tests/test-cases/Group1-Docker-Commands/1-25-Docker-Port-Map.robot

```diff
@@ -54,4 +54,58 @@ Create container without specifying host port
     Should Not Contain  ${output}  Error
     ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} stop test-redis
     Should Be Equal As Integers  ${rc}  0
-    Should Not Contain  ${output}  Error
\ No newline at end of file
+    Should Not Contain  ${output}  Error
+
+Run and exit with mapped ports
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} rm -f $(docker ${params} ps -aq)
+    Should Be Equal As Integers  ${rc}  0
+
+    ${rc}  ${output}=  Run And Return Rc And Output  mkfifo /tmp/fifo1
+    ${result}=  Start Process  docker ${params} run -i --name ctr1 -p 1900:9999 -p 2200:2222 busybox /bin/top < /tmp/fifo1  shell=True  alias=sh1
+    Sleep  5
+    ${rc}  ${output}=  Run And Return Rc And Output  echo q > /tmp/fifo1
+    ${result}=  Wait for process  sh1
+    Log  ${result.stdout}
+    Log  ${result.stderr}
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} ps -a
+    Log  ${output}
+    Should Contain X Times  ${output}  Exited  1
+
+    ${rc}  ${output}=  Run And Return Rc And Output  mkfifo /tmp/fifo2
+    ${result}=  Start Process  docker ${params} run -i --name ctr2 -p 1900:9999 -p 3300:3333 busybox /bin/top < /tmp/fifo2  shell=True  alias=sh2
+    Sleep  5
+    ${rc}  ${output}=  Run And Return Rc And Output  echo q > /tmp/fifo2
+    ${result}=  Wait for process  sh2
+    Log  ${result.stdout}
+    Log  ${result.stderr}
+    Should Be Equal As Integers  ${result.rc}  0
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} ps -a
+    Log  ${output}
+    Should Contain X Times  ${output}  Exited  2
+
+Remap mapped ports after OOB Stop
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} rm -f $(docker ${params} ps -aq)
+    Should Be Equal As Integers  ${rc}  0
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} create -it -p 10000:80 -p 10001:80 --name ctr3 busybox
+    Should Be Equal As Integers  ${rc}  0
+    Should Not Contain  ${output}  Error
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} start ctr3
+    Should Be Equal As Integers  ${rc}  0
+    Should Not Contain  ${output}  Error
+
+    # PowerOff VM out-of-band
+    ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run And Return Rc And Output  govc vm.power -off ${vch-name}/"ctr3*"
+    Run Keyword If  '%{HOST_TYPE}' == 'VC'  Should Be Equal As Integers  ${rc}  0
+    ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run And Return Rc And Output  govc vm.power -off "ctr3*"
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Should Be Equal As Integers  ${rc}  0
```

Comment:
This procedure is now done in enough places, I would like to see this as a keyword in Util.robot. Can you move that over please?